bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsEpics.scala

```diff
@@ -0,0 +1,355 @@
+package edu.gemini.seqexec.server
+
+import java.util
+import java.util.logging.Logger
+import edu.gemini.seqexec.server.TcsEpics._
+import edu.gemini.seqexec.server.tcs.{BinaryYesNo, BinaryOnOff}
+
+import collection.JavaConversions._
+
+import edu.gemini.epics.acm._
+
+import scala.concurrent.duration.Duration
+import scalaz._
+import Scalaz._
+import scalaz.concurrent.Task
+
+/**
+ * TcsEpics wraps the non-functional parts of the EPICS ACM library to interact with TCS. It has all the objects used
+ * to read TCS status values and execute TCS commands.
+ *
+ * Created by jluhrs on 10/1/15.
+ */
+
+final class TcsEpics private () {
+  import TcsEpics._
+  import EpicsCommand.setParameter
+
+
+  // This is a bit ugly. Commands are triggered from the main apply record, so I just choose an arbitrary command here.
+  // Triggering that command will trigger all the marked commands.
```

Comment:
This needs a bit more explanation. In EPICS the CAD records are used to execute commands (with or without parameters). There are also apply records, used to trigger several commands at once. Apply records can be cascaded, and in each Gemini EPICS system there is a top level apply record. The way it is done is: first we mark the CAD we want to trigger (writing a value to a parameter input counts as marking a CAD), and then trigger the top apply record. All the marked CAD records will be eventually executed. The top apply record has a CAR record associated, which indicates the state of the commands executed.
What all means is that to trigger a command and monitor its execution, we need to trigger the top apply record and monitor the top CAR record, and that is actually what the CaCommandSender post method does. But if another CAD record was marked, it also will be executed. That fact is used, for example, in the implementation of TcsControllerEpics.config. There is a series of calls that write parameters on different TCS CAD records, followed by a post that triggers the top apply. That relationship between commands and the top apply is not represented in the code.